### PR TITLE
Allow adding a song to multiple playlists at once.

### LINKED ISSUE #609: Feature request: add song to multiple playlists
Hi!
If you would like to add song to multiple playlists, it will be more user-friendly IMHO to perform this action in "One-click". Is it possible to add this feature? 

Thanks, love this project!

BR, Eugene

### PR DESCRIPTION
**Pull request by YashJipkate**
This PR introduces a new feature - add a song to multiple playlists at once.

Closes #609.  

I've added some screen recordings to cover the possible use cases:

- Add songs to two existing playlists

https://user-images.githubusercontent.com/34203227/113914309-0a47df80-97fb-11eb-90ab-56ab5401654c.mp4

- Add duplicate songs to the playlist(s)

https://user-images.githubusercontent.com/34203227/113914417-277cae00-97fb-11eb-89ce-60bc72705feb.mp4

- Add songs to a new playlist.

https://user-images.githubusercontent.com/34203227/113915118-01a3d900-97fc-11eb-940d-fe1546bcf3c8.mp4

- Add and remove playlists from the selection.

https://user-images.githubusercontent.com/34203227/113915317-4465b100-97fc-11eb-9f25-b1f1154986e9.mp4

### CODE REVIEW

**Comment by deluan** [comment]
Thanks! Looks good, will do a throughly review in the next few days!

File: ui/src/dialogs/SelectPlaylistInput.js [hunk 1]
```diff
@@ -81,7 +92,17 @@ export const SelectPlaylistInput = ({ onChange }) => {
         // Regular option
         return option.name
       }}
-      renderOption={(option) => option.name}
+      renderOption={(option, { selected }) => (
+        <React.Fragment>
+          <Checkbox
+            icon={icon}
+            checkedIcon={checkedIcon}
+            style={{ marginRight: 8 }}
```
Comment (deluan) [inline ui/src/dialogs/SelectPlaylistInput.js:100]: Please use `makeStyle()`

**Review by deluan — COMMENTED** [review]
Thanks, looks good but I'll have to do some manual testings. It would be great to have some tests as this code is getting too complicated....

File: ui/src/dialogs/SelectPlaylistInput.js (hunk 1 above)
Comment (YashJipkate) [reply ui/src/dialogs/SelectPlaylistInput.js]: Done. Will be soon adding the tests.

**Comment by YashJipkate** [comment]
Hey @deluan, I have added the tests. Few points to note:
- I have to add two new devdepedencies to make this work - `ra-test` and `jest-environment-jsdom-sixteen`, the latter of which is required because of an old version of JSDom.
- I took the liberty to refactor the `httpClient` into `dataProvider.getOne` as discussed in our Discord conversation, it was also necessary for the mocking.
- I have left out the case when the added songs are duplicate in the playlist, since that is out of the scope of this PR (This PR is already has a lot of changes) and also capturing a modal is difficult in jest (maybe a separate test for the duplicateSong dialog box).

**Comment by deluan** [comment]
Looks good! Will try to review before Monday, so it can be included in the next release.

**Pull request merged**